_posts/2020-10-19-torch-image-classification/torch-image-classification.Rmd

```diff
@@ -0,0 +1,543 @@
+---
+title: "Classifying images with torch"
+description: > 
+ We learn about transfer learning, input pipelines, and learning rate schedulers, all while using torch to tell apart species of beautiful birds.
+author:
+  - name: Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydanatorchimageclassification
+date: 10-19-2020
+categories:
+  - Torch
+  - R
+  - Image Recognition & Image Processing
+bibliography: bibliography.bib
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/image_classif_birds.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+In recent posts, we've been exploring essential `torch` functionality: [tensors](https://blogs.rstudio.com/ai/posts/2020-10-01-torch-network-from-scratch/), the sine qua non of every deep learning framework; [autograd](https://blogs.rstudio.com/ai/posts/2020-10-05-torch-network-with-autograd), `torch`'s implementation of reverse-mode automatic differentiation; [modules](https://blogs.rstudio.com/ai/posts/2020-10-07-torch-modules), composable building blocks of neural networks; and [optimizers](https://blogs.rstudio.com/ai/posts/2020-10-09-torch-optim/), the -- well -- optimization algorithms that `torch` provides.
+
+But we haven't really had our "hello world" moment yet, at least not if by "hello world" you mean the inevitable *deep learning experience of classifying pets*. Cat or dog? Beagle or boxer? Chinook or chihuahua? We'll distinguish ourselves by asking a (slightly) different question: What kind of bird?
```

Comment:
```suggestion
But we haven't really had our "hello world" moment yet, at least not if by "hello world" you mean the inevitable *deep learning experience of classifying pets*. Cat or dog? Beagle or boxer? Chinook or Chihuahua? We'll distinguish ourselves by asking a (slightly) different question: What kind of bird?
```

Reply:
Dog breed capitalization is breed-dependent, and Chihuahua gets capitalized because it's derived from a proper noun (Chihuahua, Mexico). https://thatproofreaderguy.com/are-dog-breed-names-capitalized/

Reply:
oh hehe, good to know ...that big style manual arrived but i haven't quite gotten to studying it yet :-)))